src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java

```diff
@@ -40,28 +45,47 @@ public Report parse(final ReaderFactory reader) throws ParsingException {
         }
     }
 
+    private List<String> readLogInfo(final Iterator<String> lineIterator) {
+        List<String> logInfo = new ArrayList<>();
+
+        while (lineIterator.hasNext()) {
+            String currentLine = lineIterator.next();
+            Matcher logBeginningMatcher = LOG_BEGINNING_PATTERN.matcher(currentLine);
+            if (logBeginningMatcher.matches()) {
+                logInfo.add(currentLine);
+            }
+            else {
+                int lastIdx = logInfo.size() - 1;
```

Comment:
What happens if `logInfo` is empty here? Then the next statements need to be skipped.